tests/conftest.py

```diff
@@ -26,3 +29,52 @@ def setup():
     if path.exists(config.config_location):
         remove(config.config_location)
         print('removed')
+
+
+#: hook for making test result available in fixtures
+#: ref: https://docs.pytest.org/en/latest/example/simple.html#making-test-result-information-available-in-fixtures
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    outcome = yield
+    report = outcome.get_result()
+
+    #: when could be "setup", "call", or "teardown"
+    setattr(item, f'{report.when}_report', report)
+
+
+@pytest.fixture()
+def test_gdb(request, tmp_path):
+    #: get test name and directory and look for gdb or zip
+    module = '/'.join(request.module.__name__.split('.')[1:])
+    data_folder = Path(request.module.__file__).parent / 'data' / module
+    test_name = request.function.__name__
+    original_gdb = data_folder / f'{test_name}.gdb'
+    original_zip = data_folder / f'{test_name}.zip'
+
+    if original_gdb.exists():
+        yield str(copytree(original_gdb, tmp_path / original_gdb.name))
+    elif original_zip.exists():
+        with ZipFile(original_zip) as zip_file:
+            zip_file.extractall(tmp_path)
+            yield str(tmp_path / f'{request.function.__name__}.gdb')
+    else:
+        yield None
```

Comment:
**issue**: this should throw so the user knows their gdb and function name did not match and no gdb was found before weird stuff happens with expectations in the test.

Reply:
154f31a